### PR TITLE
Minor improvements done while studying map/reduce implementation

### PR DESCRIPTION
**Pull request by bruceadowns**
- corrected tests
- applied go vet recommendations
- added tab writer
- changed done chan from bool to explicit struct{}

### CODE REVIEW

**Comment by youngkin** [comment]
Wow, that was fast. I'll have a look. Thanks!

File: mapreduce/invertedindex.go [hunk 1]
```diff
@@ -25,62 +25,68 @@ func Map(input MRInput, resultChl chan MRInput, doneChl chan bool) {
...
-	doneChl <- true
+	doneChl <- struct{}{}
 }
 
 // RemoveDups removes any duplicates from MRInput.Values. Specifically, if a given word appears more than once in a
 // file, then that file's name will appear multiple times in MRInput.Values after processing by the Map() function.
-func RemoveDups(input MRInput, resultChl chan MRInput, doneChl chan bool) {
-	set := make(map[string]bool)
+func RemoveDups(input MRInput, resultChl chan MRInput, doneChl chan struct{}) {
+	set := make(map[string]struct{})
```
Comment (youngkin) [inline mapreduce/invertedindex.go:34]: I had thought `map[string]bool` was more idiomatic, so I googled it. Turns out `struct[]` now allocates zero bytes (as of July 2012) . I guess I was reading some obsolete docs. Nice change, thanks.

File: mapreduce/invertedindex.go [hunk 2]
```diff
@@ -25,62 +25,68 @@ func Map(input MRInput, resultChl chan MRInput, doneChl chan bool) {
...
 	for _, fileName := range input.Values {
-		set[fileName] = true
+		set[fileName] = struct{}{}
 	}
```
Comment (youngkin) [inline mapreduce/invertedindex.go]: ditto comment on line 32, change back to `true`

**Comment by youngkin** [comment]
Thanks for making the fixes associated with `go vet/golint` and updating the unit tests for the new file I added (butterflies.txt).

**Comment by bruceadowns** [comment]
Thanks for looking at and merging my suggestions. Also, I appreciate your feedback on the map[string]bool change.

**Pull request merged**